src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -293,4 +294,22 @@ public static BlockStateMock mockState(@NotNull Block block)
 			return new BlockStateMock(block);
 		}
 	}
+
+	@Override
+	public boolean equals(Object other) {
+		if(other == null) {return false;}
+		if(!(other instanceof BlockState)) {return false;}
+
+		BlockState otherState = (BlockState) other;
+		if(otherState.getType() != getType()) {return false;}
+		if(otherState.isPlaced() != isPlaced()) {return false;}
+		if(!isPlaced()) {return true;}
+
+		return otherState.getBlock().getLocation().equals(getBlock().getLocation());
+	}
```

Comment:
Again, not inline with what Spigot does.
Spigot compares Location and BlockData, type/isPlaced are completely ignored.

Reply:
Spigot compares the Location and BlockData. I can't compare the BlockData as it isn't implemented, so Material is as close as I can get. BlockStateMock doesn't store a BlockPosition and World, so I can't compare those separately as CraftBukkit does. Checking if isPlaced is different (returning false if the case) and then comparing Locations afterwards is the equivalent of what Spigot does in this case. Comparing the Locations checks if the World is the same as well as the BlockPosition. 
At least, as far as I can see this is the case.